jquery.serializejson.js

```diff
@@ -45,7 +45,22 @@
       if (type !== 'skip') { // ignore elements with type 'skip'
         keys = f.splitInputNameIntoKeysArray(nameWithNoType);
         value = f.parseValue(value, name, type, opts); // convert to string, number, boolean, null or customType
-        f.deepSet(serializedObject, keys, value, opts);
+
+        // Skip serialization of false values for listed types or field names
+        var skipSerialization = false;
+        if (!value || value.length == 0) {
+          var skipFromDataAttr = f.tryToFindSkipFalsyFromDataAttr(name, $form);
+          if (skipFromDataAttr === true || skipFromDataAttr === false)
+            skipSerialization = skipFromDataAttr;
+          else if (opts.skipFalsyValuesForTypes && opts.skipFalsyValuesForTypes.indexOf(type) >= 0)
+            skipSerialization = true;
+          else if (opts.skipFalsyValuesForFields && opts.skipFalsyValuesForFields.indexOf(nameWithNoType) >= 0)
+            skipSerialization = true;
+        }
```

Comment:
Please move this logic to a helper function, and use braces to make my linter happy :)